app/utils/TaxsController.scala

```diff
@@ -65,16 +64,22 @@ abstract class TaxsController extends FrontendController
 
   protected def transformation(implicit user: User, request: Request[AnyRef]): Future[Result] = {
     extractViewModel map {
-      case noATS: NoATSViewModel => Redirect(routes.ErrorController.authorisedNoAts())
-      case result: T => obtainResult(result)
+      case Right(noAts: NoATSViewModel) => Redirect(routes.ErrorController.authorisedNoAts())
+      case Right(result: T) => obtainResult(result)
+      case Left(InvalidTaxYear) => BadRequest("Request does not contain valid tax year")
```

Comment:
Where is this being tested? (Bad request and its body)

Reply:
Controller tests changed.